src/extension/token/voting/VotingERC721.sol

```diff
@@ -0,0 +1,297 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+import {Historical} from "./Historical.sol";
+import {ModularExtension} from "../../../ModularExtension.sol";
+import {IERC721A, ERC721A, ERC721AQueryable} from "@erc721a/extensions/ERC721AQueryable.sol";
+import {SafeCastLib} from "@solady/utils/SafeCastLib.sol";
+import {EIP712} from "@solady/utils/EIP712.sol";
+import {ECDSA} from "@solady/utils/ECDSA.sol";
+
+library VotingStorage {
+    /// @custom:storage-location erc7201:token.voting
+    bytes32 public constant VOTING_STORAGE_POSITION =
+        keccak256(abi.encode(uint256(keccak256("token.voting")) - 1)) & ~bytes32(uint256(0xff));
```

Comment:
we should include the token spec, erc721 if this extension is only designed for ERC721